src/js/actions/layers.js

```diff
@@ -684,15 +758,18 @@ define(function (require, exports) {
         // TODO: Dispatch optimistically here for the other modifiers, and
         // eventually remove SELECT_LAYERS_BY_INDEX.
         var dispatchPromise,
-            revealPromise;
+            revealPromise,
+            initializePromise;
             
         if (!modifier || modifier === "select") {
             payload.selectedIDs = collection.pluck(layerSpec, "id");
             dispatchPromise = this.dispatchAsync(events.document.SELECT_LAYERS_BY_ID, payload);
             revealPromise = this.transfer(revealLayers, document, layerSpec);
+            initializePromise = this.transfer(initializeLayers, document, layerSpec);
```

Comment:
are you intending for this promise not to be waited-on by the resultant `select` promise?